io.cucumber.eclipse.python/src/io/cucumber/eclipse/python/launching/BehaveMessageEndpointProcess.java

```diff
@@ -0,0 +1,250 @@
+							OutputStream outputStream = socket.getOutputStream()) {
+						int framelength;
+						byte[] buffer = new byte[1024 * 1024 * 10];
+						while ((framelength = inputStream.readInt()) > 0) {
+							if (buffer.length < framelength) {
+								buffer = new byte[framelength];
+							}
+							inputStream.readFully(buffer, 0, framelength);
+							Envelope envelope = Jackson.OBJECT_MAPPER.readerFor(Envelope.class).readValue(buffer, 0,
+									framelength);
+							try {
+								handleMessage(envelope);
+							} catch (InterruptedException e) {
+								break;
+							}
+							outputStream.write(HANDLED_MESSAGE);
+							outputStream.flush();
+							if (envelope.getTestRunFinished().isPresent()) {
+								break;
+							}
+						}
+						outputStream.write(GOOD_BY_MESSAGE);
+						outputStream.flush();
+					}
+					socket.close();
+				} catch (IOException e) {
+					// Connection closed or error
+				} finally {
+					try {
+						serverSocket.close();
+					} catch (IOException e) {
+					}
+				}
+			}
+		});
+		thread.setDaemon(true);
+		thread.start();
+	}
+	
+	public boolean isTerminated() {
+		return serverSocket.isClosed();
+	}
+
+	public void terminate() {
+		try {
+			serverSocket.close();
+		} catch (IOException e) {
+		}
+	}
+
+	@Override
+	public void addEnvelopeListener(EnvelopeListener listener) {
+		synchronized (this) {
+			if (!consumers.contains(listener)) {
+				consumers.add(listener);
+				for (Envelope envelope : envelopes) {
+					listener.handleEnvelope(envelope);
+				}
+			}
+		}
+	}
+
+	@Override
+	public synchronized void removeEnvelopeListener(EnvelopeListener listener) {
+		synchronized (this) {
+			consumers.remove(listener);
+		}
+	}
+
+	@Override
+	public boolean canDisconnect() {
+		return false;
+	}
+
+	@Override
+	public void disconnect() throws DebugException {
+		// Not implemented
+	}
+
+	@Override
+	public boolean isDisconnected() {
+		return false;
+	}
+
+	@Override
+	public boolean canResume() {
+		return isSuspended();
+	}
+
+	@Override
+	public boolean canSuspend() {
+		return !isSuspended();
+	}
+
+	@Override
+	public boolean isSuspended() {
+		return suspended;
+	}
+
+	@Override
+	public void resume() throws DebugException {
+		suspended = false;
+		synchronized (this) {
+			notifyAll();
+		}
+	}
+
+	@Override
+	public void suspend() throws DebugException {
+		suspended = true;
+	}
+	
+	/**
+	 * Adds formatter arguments to the behave command to enable Eclipse integration.
+	 * This sets the formatter and port for the socket connection.
+	 * 
+	 * @param args List of command arguments to add to
+	 */
+	public void addBehaveArguments(List<String> args) {
+		// Add the formatter argument
+		// Format: --format module_name:ClassName
+		args.add("--format");
+		args.add("behave_cucumber_eclipse:CucumberEclipseFormatter");
+		
+		// Add port via userdata (behave's -D option)
+		args.add("-D");
+		args.add("cucumber_eclipse_port=" + serverSocket.getLocalPort());
```

Comment:
This must be an ENV variable! we should move this whole block better to the BehaveProcessLauncher and let this class have a method to return the server socket port so we can pass it as BehaveProcessLauncher.#remoteConnection(int port) builder method